nchelpers/__init__.py

```diff
@@ -8,13 +9,51 @@
 import numpy as np
 from nchelpers.date_utils import resolution_standard_name, time_to_seconds, d2ss
 
-
-def cmor_filename(**components):
-    """Return a standard CMOR filename built from supplied components"""
-    template = '{variable}_{mip_table}_{model}_{experiment}_{ensemble_member}'
-    if components['time_range']:
-        template += '_{time_range}'
-    return template.format(**components)
+# Map of nchelpers time resolution strings to MIP table names, standard where possible.
+# For an explanation of the content of this map, see the discussion in section titled "MIP table / table_id" in
+# https://pcic.uvic.ca/confluence/display/CSG/PCIC+metadata+standard+for+downscaled+data+and+hydrology+modelling+data
+standard_tres_to_mip_table = {
+    '1-minute': 'subhr', # frequency std
+    '2-minute': 'subhr', # frequency std
+    '5-minute': 'subhr', # frequency std
+    '15-minute': 'subhr', # frequency std
+    '30-minute': 'subhr', # frequency std
+    '1-hourly': '1hr', # custom: neither a MIP table nor a frequency standard term
+    '3-hourly': '3hr', # frequency std
+    '6-hourly': '6hr', # frequency std
+    '12-hourly': '12hr', # custom: neither a MIP table nor a frequency standard term
+    'daily': 'day', # MIP table and frequency standard
+    'monthly': 'mon', # frequency std
+    'yearly': 'yr', # frequency std
+}
+
+
+def cmor_type_filename(extension='', **component_values):
+    """Return a filename built from supplied component values, following the a CMOR-based filename standards in
+    https://pcic.uvic.ca/confluence/display/CSG/PCIC+metadata+standard+for+downscaled+data+and+hydrology+modelling+data    .
+
+    Produces a CMOR standard filename if all and only required CMOR filename components are defined.
+    Omits any components not in list of component names. Omits any component with a None value.
+
+    Warning: This thing is no smarter than it has to be. Does not enforce required components or any rules other than
```

Comment:
Seems like this will work fine. Being dumb isn't necessarily bad... the user has been warned :)